src/io/hdf/hdf_archive.cpp

```diff
@@ -140,25 +152,37 @@ void hdf_archive::set_access_plist(boost::mpi3::communicator& comm, bool request
 #endif
 
 
-bool hdf_archive::create(const std::filesystem::path& fname, unsigned flags)
+bool hdf_archive::create(const std::filesystem::path& fname, unsigned mode_flags)
 {
   possible_filename_ = fname;
   if (Mode[NOIO])
     return true;
   if (!(Mode[IS_PARALLEL] || Mode[IS_MASTER]))
     throw std::runtime_error("Only create file in parallel or by master but not every rank!");
   close();
-  file_id = H5Fcreate(fname.c_str(), flags, H5P_DEFAULT, access_id);
+  file_cpl_ = H5Pcreate(H5P_FILE_CREATE);
+  if (file_cpl_ == H5I_INVALID_HID)
+    throw std::runtime_error("hdf_archive failed to create file create properties!");
+  file_apl_ = H5Pcreate(H5P_FILE_ACCESS);
+  if (file_apl_ == H5I_INVALID_HID)
+  {
+    H5Pclose(file_cpl_);
+    throw std::runtime_error("hdr_archive failed to create file access properties!");
+  }
+  file_id = H5Fcreate(fname.c_str(), mode_flags, file_cpl_, file_apl_);
   return file_id != is_closed;
 }
 
-bool hdf_archive::open(const std::filesystem::path& fname, unsigned flags)
+bool hdf_archive::open(const std::filesystem::path& fname, unsigned mode_flags)
 {
   possible_filename_ = fname;
   if (Mode[NOIO])
     return true;
   close();
-  file_id = H5Fopen(fname.c_str(), flags, access_id);
+  file_apl_ = H5Pcreate(H5P_FILE_ACCESS);
+  if (file_apl_ == H5I_INVALID_HID)
+    throw std::runtime_error("hdf_archive failed to create file access properties list!");
+  file_id = H5Fopen(fname.c_str(), mode_flags, access_id);
   return file_id != is_closed;
```

Comment:
I just learned about `H5I_INVALID_HID`. It seems that our `is_closed` and `H5I_INVALID_HID` are both `-1`. Probably it is time to get ride of `is_closed`? This can be done in separate PRs.

Reply:
I wanted to minimize differences by using `is_closed` at times when that made (in standalone code fragments) no logical sense. It could be fixed in a future PR, but there is a great deal in io/hdf that needs cleanup.

Reply:
For follow up https://github.com/QMCPACK/qmcpack/issues/5528